index.js

```diff
@@ -1,29 +1,60 @@
 'use strict';
 
-function createError(err, code, props) {
-    var key;
+function assign(obj, props) {
+    let key;
+    let definition;
+
+    for (key in props) {
+        definition = Object.getOwnPropertyDescriptor(obj, key);
+
+        if (definition && !definition.writable) {
+            throw new TypeError(`Cannot assign to read only property '${key}' of object '${obj}'`);
+        }
+
+        obj[key] = props[key];
+    }
+
+    return obj;
+}
 
+function createError(err, code, props) {
     if (!(err instanceof Error)) {
         throw new TypeError('Please pass an Error to err-code');
     }
 
+    if (!props) {
+        props = {};
+    }
+
     if (typeof code === 'object') {
         props = code;
-    } else if (code != null) {
-        try {
-            err.code = code
-        } catch (err) { }
+        code = undefined;
     }
 
-    if (props) {
-        for (key in props) {
-            try {
-                err[key] = props[key]
-            } catch (err) { }
-        }
+    if (code) {
```

Comment:
```suggestion
    if (code != null) {
```

In case `code` is 0 or similar.